docs/attributes.md

```diff
@@ -0,0 +1,94 @@
+### `#[StateId]`
+
+Link your event to its state(s) with the `StateId` attribute
+
+```php
+class YourEvent extends Event
+{
+    #[StateId(GameState::class)]
+    public int $game_id;
+
+    #[StateID(PlayerState::class)]
+    public int $player_id;
+}
+```
+
+### `#[AppliesToState]`
+
+Another way to link states and events; like [`StateId`](#content-stateid), but using the attributes above the class instead of on each individual id.
+
+```php
+#[AppliesToState(GameState::class)]
+#[AppliesToState(PlayerState::class)]
+class RolledDice extends Event
+{
+    use PlayerAction;
+
+    public function __construct(
+        public int $game_id,
+        public int $player_id,
+        public array $dice,
+    )
+}
+```
+
+### `#[AppliesToSingletonState]`
+
+Use the `AppliesToSingletonState` on an event class to tell Verbs that it should always be applied to a single state (`CountState`) across the entire application (as opposed to having different counts for different states).
+
+Because we're using a [singleton state](/docs/reference/states#content-singleton-states), there is no need for the event to have a `$count_id`.
+
+```php
+#[AppliesToSingletonState(CountState::class)]
+class IncrementCount extends Event
+{
+    public function apply(CountState $state)
+    {
+        $state->count++;
+    }
+}
+```
+
+### `#[AppliesToChildState]`
+
+Use the `AppliesToChildState` on an event class to tell Verbs both the event's state and it's state's parent, useful in cases where multiple states rely on the same event.
```

Comment:
I think the use-case for this needs to be more clear. It's meant for nested states, but I honestly don't have a good example to use right now…